src/client.rs

```diff
@@ -1204,9 +1254,9 @@ impl<S, T> Drop for Client<S, T> {
 
         // Dirty shutdown
         // TODO: refactor, this is not the best way to handle state management.
-        self.stats.client_disconnecting(self.process_id);
```

Comment:
@magec Call `self.stats.disconnect()` here, and I think that should fix the clients sticking around bug.

Reply:
Added some tests (one working and another failing because of this).

Reply:
Mostafa solved this for servers using TCP timeouts. Maybe we can add them to client sockets as well?

Reply:
I think this might be caused by not having keepalives set on client sockets. 
@magec how abrupt is the connection termination? Do you allow the client to send proper TCP termination packets to PgCat or does the client just ... go away?

Reply:
You can try to apply the [function](https://github.com/postgresml/pgcat/blob/main/src/messages.rs#L561-L579) Lev mentioned in your branch and re-run your test to verify that the cause of the leak is indeed TCP parameters. 

If they are, we can move forward with your PR and file another one for client keepalives

Reply:
The client sends a TCP FIN, well, the kernel, when it kills the process, but tokio keeps blocked reading, will try this keepalive thing.